### PR TITLE
fix(chat): auto scroll to the end of the chat, hide older messages in…

### PR DESCRIPTION
**Pull request by AlexZakablukov**
…fo block

**What this PR does** 📖
auto scroll to the end of the chat, hide older messages info block

**Which issue(s) this PR fixes** 🔨
[AP-1505](https://satellite-im.atlassian.net/browse/AP-1505)

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

**Comment by phillsatellite** [comment]
Clayton found a bug and I was also able to replicate seems when you switch between chats it will start at top of convo and blue bar (you are viewing older messages) does not appear until the user scrolls a little. This didnt seem to be able to be replicate each time

https://user-images.githubusercontent.com/93608357/167697471-26daaa16-035c-49d8-a835-e37bef249840.mov

**Review by molimauro — APPROVED** [review]

**Comment by phillsatellite** [comment]
After retest I was still able to replicate bug of chat starting at the top, I had to switch between chats 2-3 times to get bug to appear 
https://user-images.githubusercontent.com/93608357/168095401-acce45a6-3432-40cd-bebb-1f69b20689c6.mov

**Comment by AlexZakablukov** [comment]
I hope I fixed it, test it one more time, please

File: components/ui/Chat/Scroll/Scroll.vue [hunk 1]
```diff
@@ -47,8 +47,7 @@ export default Vue.extend({
   data() {
     return {
       newMessageAlert: false,
-      scrollContainerObserver: null,
-      scrollContentObserver: null,
```
Comment (molimauro) [inline components/ui/Chat/Scroll/Scroll.vue:51]: I put 2 observers in the past in order to check both the scrolling element and its container. For example with only the scrolling element observer if we enlarge the chatbar by writing multiple lines of text we broke the chat scroll

**Comment by phillsatellite** [comment]
@AlexZakablukov thank you for all your hard work on this so far 😄  After retest on Chrome and Brave it worked flawlessly, on Safari it would work a little better but for some reason wouldnt go all the way to the bottom like the other browsers did. Other than the little Safari issue PR is awesome 💥 

https://user-images.githubusercontent.com/93608357/168337853-54ddbc3b-fa17-475e-806b-965eadc35fc7.mov

File: components/ui/Chat/Scroll/Scroll.vue (hunk 1 above)
Comment (AlexZakablukov) [reply components/ui/Chat/Scroll/Scroll.vue:51]: this.scrollObserver = scrollObserver.observe(
  this.$refs.scrollRef,
  this.$refs.scrollContent,
)

but they do the same so I put them both in one observer

Comment (molimauro) [reply components/ui/Chat/Scroll/Scroll.vue:51]: Oh if i check the docs i see that `observe` takes only an arg: https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver/observe

Comment (AlexZakablukov) [reply components/ui/Chat/Scroll/Scroll.vue:51]: So we can do something like this : 

```
mounted() {
    this.scrollObserver = new ResizeObserver(() => {
      // Autoscroll to the bottom only if the user is at the bottom of the chat
      !this.ui.isScrollOver && this.autoScrollToBottom()
    })
    this.scrollObserver.observe(this.$refs.scrollRef)
    this.scrollObserver.observe(this.$refs.scrollContent)
  },
  beforeDestroy() {
    if (this.scrollObserver) this.scrollObserver.disconnect()
  }
```

Comment (molimauro) [reply components/ui/Chat/Scroll/Scroll.vue:51]: Yes, thats perfect! ty :)

**Comment by molimauro** [comment]
@AlexZakablukov did you push your updates as per the comment? :)

**Comment by AlexZakablukov** [comment]
> @AlexZakablukov did you push your updates as per the comment? :)

yea

**Comment by molimauro** [comment]
Thanks!

**Review by molimauro — APPROVED** [review]

**Review by stavares843 — APPROVED** [review]

**Pull request merged**